python/ml4ir/base/model/scoring/monte_carlo_scorer.py

```diff
@@ -99,12 +91,171 @@ def __init__(
+                MonteCarloInferenceKey.NUM_TRAINING_TRIALS]
+            # Adding 1 here to account of the extra inference run with training=False.
+            self.monte_carlo_training_trials_tf = tf.constant(self.monte_carlo_training_trials + 1, dtype=tf.float32)
+
+        self.use_fixed_mask_in_testing = bool(self.model_config[MonteCarloInferenceKey.MONTE_CARLO_TRIALS].get(
+            MonteCarloInferenceKey.USE_FIXED_MASK_IN_TESTING, False))
+        if not self.use_fixed_mask_in_testing:
+            self.monte_carlo_test_trials = self.model_config[MonteCarloInferenceKey.MONTE_CARLO_TRIALS][
+                MonteCarloInferenceKey.NUM_TEST_TRIALS]
+            # Adding 1 here to account of the extra inference run with training=False.
+            self.monte_carlo_test_trials_tf = tf.constant(self.monte_carlo_test_trials + 1, dtype=tf.float32)
+
+        self.features_with_fixed_masks = self.model_config[MonteCarloInferenceKey.MONTE_CARLO_TRIALS].get(MonteCarloInferenceKey.FEATURES_WITH_FIXED_MASKS, [])
+        if self.features_with_fixed_masks:
+            self.features_with_fixed_masks = [f.strip() for f in self.features_with_fixed_masks.split()]
+            self.fixed_feature_count = len(self.features_with_fixed_masks)
+            self.fixed_mask = list(itertools.product([0, 1], repeat=self.fixed_feature_count))
+            self.monte_carlo_fixed_trials_count = tf.constant(len(self.fixed_mask), dtype=tf.float32)
+
+    def mask_inputs(self, inputs, mask_count):
+        """
+        Apply predefined masks to input features in a vectorized manner.
+
+        Parameters
+        ----------
+        inputs : dict of tf.Tensor
+            Dictionary containing input feature tensors. Each key represents the feature name, and each value is a tensor of shape [batch_size, feature_length].
+        mask_count : int
+            Number of different masks to be applied. This corresponds to the number of Monte Carlo trials.
+
+        Returns
+        -------
+        masked_inputs_list : list of dict of tf.Tensor
+            A list where each element is a dictionary of masked input feature tensors. Each dictionary has the same keys as the input dictionary, and the values are tensors of the same shape, but with masks applied.
+        """
+        # Apply masks in a vectorized manner
+        masked_inputs_list = []
+        for i in range(mask_count):
+            masked_inputs = {}
+            for key, value in inputs.items():
+                masked_value = value
+                for j in range(self.fixed_feature_count):
+                    if key == self.features_with_fixed_masks[j]:
+                        mask = tf.constant(self.fixed_mask[i][j], dtype=tf.float32)
+                        masked_value = tf.multiply(masked_value, mask)
+                masked_inputs[key] = masked_value
+            masked_inputs_list.append(masked_inputs)
+        return masked_inputs_list
```

Comment:
@mohazahran I don't understand why this mask function is here. It should be in QueryFeatureMask, right? You are modifying what the MCScorer does here.